modules/REST/V2/Languages.php

```diff
@@ -0,0 +1,737 @@
+					array( 'status' => 400 )
+				);
+			}
+
+			$language = $languages[ $request['locale'] ];
+			$defaults = array(
+				'name'       => $language['name'],
+				'slug'       => $language['code'],
+				'rtl'        => (int) ( 'rtl' === $language['dir'] ),
+				'flag'       => $language['flag'],
+				'term_group' => 0,
+			);
+		} else {
+			$defaults = array(
+				'term_group' => 0,
+			);
+		}
+
+		$prepared = (array) $this->prepare_item_for_database( $request );
+		$prepared = array_merge( $defaults, $prepared );
+
+		/**
+		 * @phpstan-var array{
+		 *     locale: non-empty-string,
+		 *     name: string,
+		 *     slug: string,
+		 *     rtl: 0|1,
+		 *     flag: string,
+		 *     term_group: 0|1
+		 * } $prepared
+		 */
+		$result = $this->model->languages_model->add( $prepared );
+
+		if ( is_wp_error( $result ) ) {
+			return $result;
+		}
+
+		/** @var PLL_Language */
+		$language = $this->model->languages_model->get( $prepared['slug'] );
+		$response = $this->prepare_item_for_response( $language, $request );
+
+		return rest_ensure_response( $response );
+	}
+
+	/**
+	 * Retrieves one language from the collection.
+	 *
+	 * @since 3.7
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 *
+	 * @phpstan-template T of Fields
+	 * @phpstan-param WP_REST_Request<T> $request
+	 */
+	public function get_item( $request ) {
+		$language = $this->get_language( $request['code'] );
+
+		if ( is_wp_error( $language ) ) {
+			return $language;
+		}
+
+		$response = $this->prepare_item_for_response( $language, $request );
+
+		return rest_ensure_response( $response );
+	}
+
+	/**
+	 * Updates one language from the collection.
+	 *
+	 * @since 3.7
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 *
+	 * @phpstan-template T of Fields
+	 * @phpstan-param WP_REST_Request<T> $request
+	 */
+	public function update_item( $request ) {
+		$language = $this->get_language( $request['code'] );
+
+		if ( is_wp_error( $language ) ) {
+			return $language;
+		}
+
+		$prepared = (array) $this->prepare_item_for_database( $request );
+
+		if ( ! empty( $prepared ) ) {
+			if ( ! isset( $prepared['lang_id'], $prepared['locale'], $prepared['name'], $prepared['rtl'], $prepared['term_group'] ) ) {
+				$prepared = array_merge(
+					array(
+						'lang_id'    => $language->term_id,
+						'locale'     => $language->locale,
+						'name'       => $language->name,
+						'rtl'        => $language->is_rtl,
+						'term_group' => $language->term_group,
+					),
+					$prepared
+				);
+			}
+			/**
+			 * @phpstan-var array{
+			 *     lang_id: int,
+			 *     locale: string,
+			 *     name: string,
+			 *     slug: string,
+			 *     rtl: 0|1,
+			 *     term_group: int
+			 * } $prepared
+			 */
```

Comment:
Because:

```
Parameter #1 $args of method WP_Syntex\Polylang\Models\Languages::update() expects array{lang_id: int|numeric-string, name: string, ...}, non-empty-array given.
```

It seems difficult to properly tag `prepare_item_for_database()`.